### PR TITLE
test salt-jenkins with travis

### PR DESCRIPTION
**Pull request by gtmanfred**
Fixes saltstack/release-mgmt#59

### CODE REVIEW

File: locale.sls [hunk 1]
```diff
@@ -9,6 +9,15 @@
...
     - name: glibc-locale
+{% elif grains.os_family == 'Debian' %}
+deb_locale:
+  pkg.installed:
+    - pkgs:
+      - locales
+      - console-data
+      - dbus
+  service.running:
+    - name: dbus.socket
```
Comment (rallytime) [inline locale.sls]: 👍

File: .kitchen.yml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    provisioner:
+      pillars:
+        top.sls:
+          base:
+            "*":
+              - jenkins
+        jenkins.sls:
+          test_git_url: git://github.com/saltstack/salt.git
+          test_git_commit: <%= (ENV['TRAVIS_BRANCH'] == 'master' ? 'develop' : ENV['TRAVIS_BRANCH']) %>
+          py3: true
```
Comment (rallytime) [inline .kitchen.yml]: I haven't looked into your good work with kitchen-salt yet, but for the Python 3 tests to run in Jenkins, the `state.sls git.salt` call also needs `pillar="{py3: true}"` on the CLI call. Is that essentially what this line does as well?

Comment (gtmanfred) [reply .kitchen.yml]: This line checks which branch of salt/git to use for cloning to /testing.

this line enables py3 tests https://github.com/saltstack/salt-jenkins/pull/514/files#diff-d3fbd21bc6be7ce4531c27a82b2619deR86

Comment (gtmanfred) [reply .kitchen.yml]: This is the list of iterations for the platforms and the suites.

```
$ bundle exec kitchen list
Instance         Driver  Provisioner  Verifier  Transport  Last Action    Last Error
py2-centos-7     Docker  SaltSolo     Busser    Ssh        <Not Created>  <None>
py2-centos-6     Docker  SaltSolo     Busser    Ssh        <Not Created>  <None>
py2-ubuntu-1604  Docker  SaltSolo     Busser    Ssh        <Not Created>  <None>
py2-debian-8     Docker  SaltSolo     Busser    Ssh        <Not Created>  <None>
py2-debian-9     Docker  SaltSolo     Busser    Ssh        <Not Created>  <None>
py2-arch         Docker  SaltSolo     Busser    Ssh        <Not Created>  <None>
py2-opensuse     Docker  SaltSolo     Busser    Ssh        <Not Created>  <None>
py3-centos-7     Docker  SaltSolo     Busser    Ssh        <Not Created>  <None>
py3-centos-6     Docker  SaltSolo     Busser    Ssh        <Not Created>  <None>
py3-ubuntu-1604  Docker  SaltSolo     Busser    Ssh        <Not Created>  <None>
py3-debian-8     Docker  SaltSolo     Busser    Ssh        <Not Created>  <None>
py3-debian-9     Docker  SaltSolo     Busser    Ssh        <Not Created>  <None>
py3-arch         Docker  SaltSolo     Busser    Ssh        <Not Created>  <None>
py3-opensuse     Docker  SaltSolo     Busser    Ssh        <Not Created>  <None>
```

Comment (rallytime) [reply .kitchen.yml]: Ok thanks.

**Comment by rallytime** [comment]
It looks to me here like you're running the bootstrap script to install salt before running the salt-jenkins states. If we're running this on containers anyway, can salt be installed already on the containers, then just run the salt-jenkins states?

That way we can test just the state runs, rather than the salt installation step?

Though, now that you have this working, it might be a good way to test the bootstrap script itself. I've been wanting to add something like that to bootstrap PRs for a while, just haven't had the time  - this setup would work well!

**Comment by gtmanfred** [comment]
We could use containers that already have salt installed, but part of kitchen-salt is to setup and install salt

https://github.com/saltstack/kitchen-salt/blob/master/provisioner_options.md#salt_install

Those are the different methods that could be used.

**Comment by gtmanfred** [comment]
Also, once the provisioner passes, I am going to setup `testinfra` to login and make sure that python3 or python2 can run and that the correct git repo branch is available in /testing.

**Comment by rallytime** [comment]
Yes, very cool. Thanks for those links. I'll look into that for the salt-bootstrap PR tests. We don't get a TON of PRs over there, so I think travis will work there as well.

**Comment by rallytime** [comment]
YAY!!!

**Pull request merged**